openhands-sdk/openhands/sdk/conversation/visualizer/base.py

```diff
@@ -0,0 +1,87 @@
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, final
+
+from openhands.sdk.event.base import Event
+
+
+if TYPE_CHECKING:
+    from openhands.sdk.conversation.base import ConversationStateProtocol
+    from openhands.sdk.conversation.conversation_stats import ConversationStats
+
+
+class ConversationVisualizerBase(ABC):
+    """Base class for conversation visualizers.
+
+    This abstract base class defines the interface that all conversation visualizers
+    must implement. Visualizers can be created before the Conversation is initialized
+    and will be configured with the conversation state automatically.
+
+    The typical usage pattern:
+    1. Create a visualizer instance:
+       `viz = MyVisualizer(name="agent1")`
+    2. Pass it to Conversation: `conv = Conversation(agent, visualizer=viz)`
+    3. Conversation automatically calls `viz.initialize(state)` to attach the state
+
+    You can also pass the uninstantiated class if you don't need extra args
+        for initialization, and Conversation will create it:
+         `conv = Conversation(agent, visualizer=MyVisualizer)`
+    Conversation will then calls `MyVisualizer()` followed by `initialize(state)`
+    """
+
+    _name: str | None
+    _state: "ConversationStateProtocol | None"
+
+    def __init__(
+        self,
+        name: str | None = None,
+    ):
+        """Initialize the visualizer base.
+
+        Args:
+            name: Optional name to prefix in panel titles to identify
+                                  which agent/conversation is speaking. Will be
+                                  capitalized automatically.
+        """
+        self._name = name.capitalize() if name else None
+        self._state = None
+
+    @final
+    def initialize(self, state: "ConversationStateProtocol") -> None:
+        """Initialize the visualizer with conversation state.
+
+        This method is called by Conversation after the state is created,
+        allowing the visualizer to access conversation stats and other
+        state information.
+
+        Subclasses can override this method to add custom initialization logic,
+        but should call super().initialize(state) to ensure the state is set.
+
```

Comment:
```suggestion
        Subclasses should not override this method, to ensure the state is set.

```
Nit: since we marked as final, small adjustment with our expectation